### PR TITLE
Double Solar Output

### PR DESCRIPTION
**Pull request by Deanthelis**
This PR is straightforward: Solars will produce twice as much power - a maximum of 180kW per array, for 720kW if all four are wired and powered (hypothetically, in practice it does fluctuate).

What this means:
-Three solar arrays can power the station completely. Currently, even all four can't run it forever - the APC's starve, albeit slowly.
-Four active solar arrays can power the station + produce enough excess to start up and maintain singularity containment. Ergo, an engineering borg dragging a cable coil around can power the station all by itself, as can maintenance drones, in a reasonable amount of time, no hands required.

I did a bit of local testing with the changes made. One solar array, with the Engineering SMES entirely off and the array wired directly into the grid, will buy about 15 or 20 minutes before the APC's start dying off. Wiring in a second buys you another half hour to an hour, depending on what's being done. Three solars wired will power the whole station, and a fourth solar wired gives you some extra juice to charge SMES with or fire up the singularity.

TL;DR: Solars are an actual valid power source with a reasonable setup time now, as they should have been forever ago. Praise the Sun.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
:-1: 

You should have to start and maintain the singularity to actually get to use power gloves and maintain the network at that level---likewise, solar arrays are borked at the moment and don't properly update, so getting an accurate result for this is impossible and this really shouldn't be messed with until that bug is fixed.

Arrays are more than capable of producing 90,000+ kilowatts per array when the sun is directly hitting them---and it really only requires ~100,000-150,000 kilowatts to keep the station from running out of power.

**Comment by Deanthelis** [comment]
Okay. Question time.
1: Why should the Singularity be the only thing capable of making Power Gloves useful? If that's the case, remove PACMAN generators. A single upgraded PACMAN can produce the power level of two solars currently, and can do so for about 45 minutes. This means Science can do Engineering better than Engineering can in emergencies.
2: Why should the means of powering the station beyond the absolute minimum be regulated by a singular traitor item? Why aren't you regulating it from the much more common sources of powernet-based damage, shocked airlocks and exposed grilles?
3: Why do you think the difference between 360kW and 720kW is significant, in the case of power gloves or a shocked airlock/vendor/grille/etc?
4: Why do you keep saying kilowatts when a kilowatt is a thousand watts? 
5: When was the last time you saw someone use power gloves on Paradise to any great effect?

**Comment by Fox-McCloud** [comment]
- (1) there's risk to the singularity and general setup time to get it up and running and stable at a rate at which it'll produce enough power to arcflash things or turn powergloves into deathgloves--solars is just wire and forget--it's self sustaining
- (2) Because the big picture is you don't need more than ~150,000 watts to power the entire station.
- (3) Because of how shock damage scales; it operates in steps as opposed to a linear scales, meaning going from one step to the next has a *significant* effect, even if it's only a 1 watt difference from one step to the next.
- (4) Habit, really; some stations use kilowatts instead of watts to make the numbers seem more impressive; I tend to use the two interchangably (despite not being so). 

And again
- Solars are broken at the moment and don't track the sun properly, skewing results of true solar output
- Again, it only requires ~150,000 watts, at best to power the station; anything else is overkill/traitor related/having fun to see how high you can push that number/popping powersinks.

**Comment by Deanthelis** [comment]
1: I don't know when you last decided to wire solars, but it takes a considerable amount of time. Moreso if the player isn't perfectly efficient (a reasonable assumption) and/or has any amount of lag (also a reasonable assumption). If you're by yourself, the time spent wiring them and moving to the next array can be upwards of 30 minutes to do them all - rather close to the singularity's setup time. When testing locally, it took me about 25 minutes - and that's with no lag, and doing it without making any unnecessary knots, as well as admin-abusing the necessary equipment instead of having to gear up. 
1a: Solars, with this change, are incapable of making powergloves into deathgloves, just as they are now, so that's irrelevant to the discussion. They also cannot arcflash things - in fact, arcflashing takes such a massive amount of power that it only happens once a week, at BEST, so that's also irrelevant to the discussion.
1b: This change doesn't in any way propose that solars should rival the singularity, only that they should do more than the bare minimum.

2: That doesn't answer the question at all. I asked why you're balancing power production around an item that only exists in only a couple gamemodes and is rarely used, and not the most common sources of powernet damage. 

3: There is literally no difference when it comes to damage between 360kW and 720kW. They both deal the same amount:

/datum/powernet/proc/get_electrocute_damage()
	switch(avail)
		//Rest of the switch block
		if (200000 to 1000000)
			return min(rand(25,80),rand(25,80))

The fact that damage isn't calculated directly from the available power but is instead set in steps has nothing to do with this change. I don't know why you brought that up. It would only matter if solar output went beyond a million watts, which it doesn't.

4: Fair enough, I guess, but we're discussing Paradise here, not other non-downstream stations.

5: Way to just utterly ignore this question.

Solar arrays currently have an absolute maximum of 90kW. That is a hard, mathematical maximum; when the bug is fixed, it will fluctuate between that and zero. The only 'skewing' is downward, which only helps my point here.

This discussion is also completely divorced from whether or not they work, because increasing their output or keeping their output where it is relies on them working. I don't know why you brought that up again, I thought I made it pretty clear that it's irrelevant. This isn't a big, complicated change to the way solars work; there's no reason not to make this change prior to the bugfix. And, as far as I could tell when testing locally, the bug just prevents the solars from aligning automatically. You can still manually align them (which makes them not so wire-and-forget).

**Comment by Fox-McCloud** [comment]
I'm not here to argue at length with you. The fact of the matter is, these don't function currently, so it's pointless *to* argue because currently power generation that is done by them is not functioning properly due to a bug that has yet to be pinned down. Until that bug is fixed and we see what *actual* power output is by four functional solar arrays, then all arguments regarding this are moot. It would be like if the singularity didn't charge collectors properly and making the argument that "the collectors don't produce enough power; let's double it!" There's no way of knowing what the actual power output of the singularity *really* is until that's fixed and new data is generated.

**Comment by Earthdivine** [comment]
A small bit of info to throw out, the arrays are no longer able to power the station at all since the fix to lights taking power. There are plenty of engineering people who refuse to use the singlo because they hate the massive risk involved in it, why should they be forced to use this high risk and very low reward (infinate power forever is kinda bland) because of issues with Powergloves? In fact anyone using Powergloves wouldn't use the arrays, they would hotwire the engine and achieve far more power then the arrays could ever output.

**Comment by Fox-McCloud** [comment]
Using a hacky method to get solars working (on a limited basis)...Direct-wired, they produced this much power: ~300,000. Station load (while a lot of APCs were charging): ~250,000.  Station load (when APCs are charged): ~155,000.

This was done with the solars direct wired and all station SMESes turned off; the only power, at all, coming in was from the solars.

Even when fixed, solars do NOT need a buff.

The test results do fluctuate through the shift, as the sun moves and falls within the station's shadow and what have you, but it's not going to stay in that zone for an indefinite period and...APCs very much have internal batteries to hold them over until the sun comes back 'round again.

**Comment by Taaketa** [comment]
Basically the point you are both making is fix the solar tracking first. Then see if solar cells need sn increase.

**Comment by KasparoVy** [comment]
Something needs to be fixed but we need Sherlock Holmes to figure out just what it is

**Comment by Taaketa** [comment]
Why is it a difficult problem to figure out? I was gonna hop tonight and wire the solars. See if I can optimise them like Fox Mccloud says.

**Comment by Taaketa** [comment]
I tried messing with the solar arrays last night. Could never get any of them to produce above 90000 watts at any one time. So I'm not sure if I'm doing anything wrong but tried a few different configurations.

The best I could get was 90000 watts by themselves and usually only 2 solar arrays are producing that at the same time. Whilst the solars their power decreases to somewhere between 1000 - 40000 watts when the station falls into shadow.

This doesn't seem to keep the station charged in live. But I also ran in local and the charge wasn't a problem. I had the latest version of the server downloaded from Git and worked fine.

**Comment by ZomgPonies** [comment]
Balancing a buggy feature is a bad idea. :-1:

**Comment by Fox-McCloud** [comment]
Going to close this; a new PR can be cre-created after the bugs/issues with solars have been resolved and players have had a chance to see how well the station handles with just running on them.
